File: example/typescript/async/async.ts
const foo0 = async () => 42;

foo0().then(e => console.log(e))  // 42

const deffer = new Promise((resolve, reject) => setTimeout(_ => resolve(2), 2000))

async function fooDeffer() {
  console.log(1)
  console.log(await deffer)
  console.log(3)
}

fooDeffer()

File: example/dart/async/async.dart
import 'dart:async';

Future<int> foo0() async => 42;

Future<num> deffer = new Future.delayed(Duration(seconds: 2), () {
  return 2;
});

fooDeffer() async {
  print(1);
  print(await deffer);
  print(3);
}

main(List<String> args) {
  foo0().then((e) => print(e));  // 42

  fooDeffer();
}